eth/catalyst/api.go

```diff
@@ -435,8 +435,15 @@ func (api *ConsensusAPI) forkchoiceUpdated(update engine.ForkchoiceStateV1, payl
 	// sealed by the beacon client. The payload will be requested later, and we
 	// will replace it arbitrarily many times in between.
 	if payloadAttributes != nil {
-		if api.eth.BlockChain().Config().Optimism != nil && payloadAttributes.GasLimit == nil {
-			return engine.STATUS_INVALID, engine.InvalidPayloadAttributes.With(errors.New("gasLimit parameter is required"))
+		if api.eth.BlockChain().Config().Optimism != nil {
+			if payloadAttributes.GasLimit == nil {
+				return engine.STATUS_INVALID, engine.InvalidPayloadAttributes.With(errors.New("gasLimit parameter is required"))
+			}
+			if api.eth.BlockChain().Config().IsHolocene(payloadAttributes.Timestamp) && payloadAttributes.EIP1559Params == nil {
+				return engine.STATUS_INVALID, engine.InvalidPayloadAttributes.With(errors.New("eip1559Params parameter is required"))
+			} else if payloadAttributes.EIP1559Params != nil {
```

Comment:
Doesn't this now also return invalid if we're at Holocene already? I mean, don't we need to do
```suggestion
			} else if !api.eth.BlockChain().Config().IsHolocene(payloadAttributes.Timestamp) && payloadAttributes.EIP1559Params != nil {
```
?

Reply:
sigh, yes, meant to nest like the one you noted below.... fixed.